pkg/kubectl-argo-rollouts/cmd/get/get.go

```diff
@@ -0,0 +1,372 @@
+package get
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/juju/ansiterm"
+	"github.com/spf13/cobra"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/info"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/viewcontroller"
+)
+
+const (
+	example = `
+  # Get a rollout
+  %[1]s get ROLLOUT
+`
+)
+
+var (
+	colorMapping = map[string]int{
+		info.IconWaiting:     FgYellow,
+		info.IconProgressing: FgBlue,
+		info.IconWarning:     FgRed,
+		info.IconUnknown:     FgYellow,
+		info.IconOK:          FgGreen,
+		info.IconBad:         FgRed,
+		info.IconPaused:      FgWhite,
+		//info.IconNeutral:     FgWhite,
+	}
+)
+
+const (
+	IconRollout    = "⟳"
+	IconRevision   = "#"
+	IconReplicaSet = "⧉"
+	IconPod        = "□"
+	IconService    = "⑃" // other options: ⋲ ⇶ ⋔ ⤨
+	IconExperiment = "Σ" // other options: ꀀ ⋃ ⨄
+	IconAnalysis   = "α" // other options: ⚯
+)
+
+// ANSI escape codes
+const (
+	escape    = "\x1b"
+	noFormat  = 0
+	Bold      = 1
+	FgBlack   = 30
+	FgRed     = 31
+	FgGreen   = 32
+	FgYellow  = 33
+	FgBlue    = 34
+	FgMagenta = 35
+	FgCyan    = 36
+	FgWhite   = 37
+	FgDefault = 39
+)
+
+type GetOptions struct {
+	watch   bool
+	noColor bool
+
+	rolloutUpdates chan *info.RolloutInfo
+
+	options.ArgoRolloutsOptions
+}
+
+// NewCmdGet returns a new instance of an `rollouts get` command
+func NewCmdGet(o *options.ArgoRolloutsOptions) *cobra.Command {
+	getOptions := GetOptions{
+		ArgoRolloutsOptions: *o,
+	}
+
+	var cmd = &cobra.Command{
+		Use:          "get ROLLOUT",
+		Short:        "Get details about a rollouts",
+		Example:      o.Example(example),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return o.UsageErr(c)
+			}
+			name := args[0]
+			controller := viewcontroller.NewController(o.Namespace(), name, getOptions.KubeClientset(), getOptions.RolloutsClientset())
+			ctx := context.Background()
+			//ctx, cancel := context.WithCancel(ctx)
+			controller.Start(ctx)
+
+			ri, err := controller.GetRolloutInfo()
+			if err != nil {
+				return err
+			}
+			if !getOptions.watch {
+				getOptions.PrintRollout(ri)
+			} else {
+				getOptions.rolloutUpdates = make(chan *info.RolloutInfo)
+				controller.RegisterCallback(getOptions.RefreshRollout)
+				go getOptions.WatchRollout(ctx.Done())
+				controller.Run(ctx)
+				close(getOptions.rolloutUpdates)
+			}
+			return nil
+		},
+	}
+	o.AddKubectlFlags(cmd)
+	cmd.Flags().BoolVarP(&getOptions.watch, "watch", "w", false, "Watch live updates to the rollout")
+	cmd.Flags().BoolVar(&getOptions.noColor, "no-color", false, "Do not colorize output")
+	return cmd
+}
+
+const (
+	tableFormat = "%-17s%v\n"
+)
+
+func (o *GetOptions) Clear() {
+	fmt.Fprint(o.Out, "\033[H\033[2J")
```

Comment:
> There's a lot of untested code in the [info package](https://codecov.io/gh/argoproj/argo-rollouts/pull/230/tree/pkg/kubectl-argo-rollouts/info), but I think it is fine for now since the code is for a command-line tool. We can circle back to it for v0.7, and thankfully a large part of the untested sections looks reasonable to test by creating mocks.

Yes. The pod and replicaset assessment code was lifted from Argo CD, so I didn't bother writing tests for them, also because the CLI test was indirectly testing them.